mailserver/mailserver.go

```diff
@@ -450,17 +388,19 @@ func (s *WMailServer) SyncMail(peer *whisper.Peer, request whisper.SyncMailReque
 // exceedsPeerRequests in case limit its been setup on the current server and limit
 // allows the query, it will store/update new query time for the current peer.
 func (s *WMailServer) exceedsPeerRequests(peer []byte) bool {
-	s.muLimiter.RLock()
-	defer s.muLimiter.RUnlock()
-
-	if s.limiter != nil {
-		peerID := string(peer)
-		if !s.limiter.isAllowed(peerID) {
-			log.Info("peerID exceeded the number of requests per second")
-			return true
-		}
-		s.limiter.add(peerID)
+	s.muRateLimiter.RLock()
+	defer s.muRateLimiter.RUnlock()
+
+	if s.rateLimiter == nil {
+		return false
+	}
+
+	peerID := string(peer)
```

Comment:
tiny nitpick, feel free to ignore

```suggestion
	if s.rateLimiter.IsAllowed(peerID) {
	    s.rateLimiter.Add(peerID)
	    return false
	}
	
	log.Info("peerID exceeded the number of requests per second")
	return true
     }
```